sktime/forecasting/base/_base.py

```diff
@@ -260,6 +257,7 @@ def fit(self, y, X=None, fh=None):
         self._is_vectorized = vectorization_needed
         # we call the ordinary _fit if no looping/vectorization needed
         if not vectorization_needed:
+            fh = self._check_fh(fh=fh, index=y_inner.index)
```

Comment:
this will be problematic, since we will still need to do this when we vectorize!

Suggest to do this directly after `_check_X_y` (line 249 or 250)  and not in the vectorization case distinction.

Reply:
I do not entirely understand the effect of `_vectorize` method, but my guess was that it takes the method in the first argument, say `fit`, rearranges the data, and runs this `fit` with it, effectively returning back to  `vectorization_needed = False`. So, `_check_fh` is still called.

Reply:
perhaps it still works, but it confounds the layers and makes the structure (and with it, future refactors) more difficult. Better leave it where it was...

Reply:
yes, a reference to the original container is in the `X` attribute, but now that you point it out it is not too good to do it that way.

Argh, I see, this gets us into coupling territory... there are two ways to die, couple the `ForecastingHorizon` with `VectorizedDF`, or do what you did.

Is there a solution that allows us to decouple `ForecastingHorizon` and `VectorizedDF`?

Reply:
how do you handle `numpy` anyway?

Reply:
> how do you handle `numpy` anyway? Staying away from `VectorizedDF` for a moment...

`_check_X_y` converts the data into "Series, Panel, or Hierarchical object, or VectorizedDF". Is it even possible to get `y_inner` or `X_inner` as numpy arrays?

Reply:
As you asked, I moved `_check_fh` back to its original position outside of the `if...else` statement. In some cases it is moved one or two lines lower just to get access to `y_inner` which appears only after `_check_X_y`.

I have also changed the arguments of `_check_fh` so that `y_inner` or `self._y` are passed instead of their indices. the problem of index extraction is transferred to `infer_freq` which now uses `singledispatch` to address objects of various types.